tuw_nlp/graph/fourlang.py

```diff
@@ -27,6 +28,68 @@ def merge_definition_graph(self, graph, node, substitute=False):
                 F.nodes[node]['expanded'] = True
             self.G = F
 
+    def append_zero_paths(self):
+        edges = []
+        for edge in self.G.edges(data=True):
+            X = edge[0]
+            Y = edge[1]
+            color = edge[2]['color']
+
+            zero_graph, nodes_to_append = self.find_zero_paths(Y)
+
+            for node in nodes_to_append:
+                if X != node:
+                    edges.append((X, node, color))
+
+            zero_graph, nodes_to_append = self.find_zero_paths(X)
+            for node in nodes_to_append:
+                node_edges = self.G.edges(node, data=True)
+                for n in node_edges:
+                    n_color = n[2]["color"]
+                    if X != n[1]:
+                        edges.append((X, n[1], n_color))
+
+        edges = list(set(edges))
+        for edge in edges:
+            self.G.add_edge(edge[0], edge[1], color=edge[2])
+
+    def find_zero_paths(self, from_node=None):
+        if not from_node:
+            from_node = self.root
+        whitelist = []
+        zero_graph = nx.DiGraph()
+        zero_graph.add_node(from_node, name=self.G.nodes[from_node]["name"])
+        delete_list = []
+        for edge in self.G.edges(data=True):
+            if not edge[2]["color"]:
```

Comment:
if color always has to be 0, 1, or 2, then better to write `if .... == 0`... it took me a while to figure this out and I'm still not sure I'm correct